Documentation/boostrap-new-os.md

```diff
@@ -0,0 +1,70 @@
+The way to solve this problem is to create a bootstrap CLI by taking an existing CLI as a "seed" and replacing native components in it by the native components that we build for the new platform. The seed CLI we use has to be for the same processor architecture as our new platform or we have to use a seed cli that we build from sources on another platform that is already supported and opt for not crossgenning the managed components. 
+### Choosing the seed CLI
+While the bootstrap CLI can be built for any version of the seed CLI, it is important to pick a version of CLI that can be used to build current corefx, coreclr and core-setup repos. From time to time, a change in the CLI causes it to not to be usable for compiling one of the repos without fixes in the msbuild project files. So to stay on the safe side, the best practice is to pick a version used by one of the three repos. This version can be found in the `DotnetCLIVersion.txt` file in the root of each repo. 
+### Getting the seed CLI
+After choosing the seed CLI version as described in the previous paragraph, it can be used to construct an URL to download the .tar.gz file with the seed CLI itself. The way to construct the URL can be found in the `init-tools.sh` file in the root of each repo. Currently, it is constructed as follows:
+```
+https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/dotnet-sdk-${__DOTNET_TOOLS_VERSION}-${__PKG_RID}-${__PKG_ARCH}.tar.gz
+```
+The `${__DOTNET_TOOLS_VERSION}` is replaced by the seed CLI version, the `${__PKG_RID}` by the RID of the current platform and `${__PKG_ARCH}` by the architecture of the current platform.
+So for tools version `2.0.0` on linux distro with x64 architecture where portable dotnet core can be used, the URL is `https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0/dotnet-sdk-2.0.0-linux-x64.tar.gz`
+To download it, `wget` or `curl` tools can be used.
+Once the file is downloaded, create a new folder and untar the file into it.  
+So e.g. for the file mentioned above, use:
+```bash
+tar -xf dotnet-sdk-2.0.0-linux-x64.tar.gz
+```
+### Choosing RID for the new OS
+The new RID represents your target OS. It is used by developers to target that OS. The RID format is `<Name>.<Version>`. The name is lower case and should match the target OS. The version is optional and should be used if it is expected that in future versions the libraries that .NET Core depends on will not be binary compatible with the first supported version. So for example for FreeBSD 11, the RID would likely be `freebsd.11`.
+For OS that has `/etc/os-release` file, the RID needs to match the `$ID.$VERSION_ID` extracted from that file. 
+### Prerequisites
+The following libraries and tools need to be installed in order to build the bootstrap CLI. The precise names of the packages are OS specific.
+#### Tools
+* Clang version 3.5 or higher, 3.9 recommended
```

Comment:
Figuring out the dependencies and package names is one of the more annoying parts of trying to bootstrap.  Coreclr documentation also has a pretty good list of dependencies in their doc.  Ultimately, sadly, I think it just boils down to try to build and then install whatever coreclr is complaining about.  The list you have here is pretty good, it's too bad you can't just link to a central shared list though.  If anything changes, this is likely going to get out of sync pretty quickly.

Reply:
This list tends to really be os and distro specific with various small variations. So it is really not possible to put down a definitive os proof and distro proof list. 
The doc in coreclr is valid only for debian / red hat related distros that we already support.